whylog/tests/tests_front/tests_whylog_factory.py

```diff
@@ -0,0 +1,13 @@
+from whylog.front.whylog_factory import whylog_factory
+from whylog.log_reader import LogReader
+from whylog.teacher import Teacher
+from whylog.tests.utils import TestRemovingSettings
+
+
+class TestWhylogFactory(TestRemovingSettings):
+    def tests_whylog_factory(self):
+        log_reader, teacher_generator = whylog_factory()
+        teacher = teacher_generator()
+
+        assert isinstance(log_reader, LogReader)
```

Comment:
Test behavior, not identity. One day someone will make the factory return ThreadedLogReader by default and then this test will fail without reason - ThreadedLogReader will keep the same interface, after all. By creating tests that rely on identity of objects, you make them fragile. Soon when modifying code you will find that by yourself.